### PR TITLE
Support value_specs for Ports

### LINKED ISSUE #2532: Enhancement: Support value_specs for ports
We have a need for using [value_specs](https://docs.openstack.org/heat/zed/template_guide/openstack.html#OS::Neutron::Port-prop-value_specs) for Ports. This is an optional map of extra parameters to send as part of the request, and it is part of the official API as seen in the link.

This has been previously proposed and discarded in https://github.com/gophercloud/gophercloud/issues/2180 but I think it is worth reconsidering since it is actually in the API. I'm opening a new issue to hopefully get some attention to this since there was no answer on the closed issue.

### PR DESCRIPTION
**Pull request by lentzi90**
Fixes #2532

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

Docs: https://docs.openstack.org/heat/zed/template_guide/openstack.html#OS::Neutron::Port-prop-value_specs

### CODE REVIEW

File: openstack/networking/v2/ports/requests.go [hunk 1]
```diff
@@ -150,14 +151,15 @@ type UpdateOptsBuilder interface {
...
-	AllowedAddressPairs *[]AddressPair `json:"allowed_address_pairs,omitempty"`
+	Name                *string           `json:"name,omitempty"`
+	Description         *string           `json:"description,omitempty"`
+	AdminStateUp        *bool             `json:"admin_state_up,omitempty"`
+	FixedIPs            interface{}       `json:"fixed_ips,omitempty"`
+	DeviceID            *string           `json:"device_id,omitempty"`
+	DeviceOwner         *string           `json:"device_owner,omitempty"`
+	SecurityGroups      *[]string         `json:"security_groups,omitempty"`
+	AllowedAddressPairs *[]AddressPair    `json:"allowed_address_pairs,omitempty"`
+	ValueSpecs          map[string]string `json:"value_specs,omitempty"`
```
Comment (pierreprinetti) [inline openstack/networking/v2/ports/requests.go:162]: I know it's horrible, but could you please change this field to be a pointer?
Before your change, `UpdateOpts` is [comparable](https://go.dev/ref/spec#Comparison_operators). Having it as a pointer would keep it comparable, and thus would not break compilation of any consumer code.

Comment (lentzi90) [reply openstack/networking/v2/ports/requests.go:162]: Good point, done!

Comment (lentzi90) [reply openstack/networking/v2/ports/requests.go:162]: Argh... messed up the branches. go mod does not like branches with `/` in the name, and I'm using the branch here for a PR in CAPO (https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1452), which means I had to change the branch name.
I will create a new PR with the correct branch name.

Comment (pierreprinetti) [reply openstack/networking/v2/ports/requests.go:162]: Ok, thanks! I am closing this one to avoid confusion.

Comment (lentzi90) [reply openstack/networking/v2/ports/requests.go:162]: The new PR is https://github.com/gophercloud/gophercloud/pull/2537. It has the same content (with the fixed the pointer). Sorry for the mess.

Ref for go issue with slashes in branch names: https://github.com/golang/go/issues/32955#issuecomment-511860346

Comment (pierreprinetti) [reply openstack/networking/v2/ports/requests.go:162]: Good to know!

**Pull request closed without merging**